homeassistant/components/freebox/router.py

```diff
@@ -89,7 +93,39 @@ async def update_all(self, now: datetime | None = None) -> None:
     async def update_device_trackers(self) -> None:
         """Update Freebox devices."""
         new_device = False
-        fbx_devices: list[dict[str, Any]] = await self._api.lan.get_hosts_list()
+
+        fbx_devices: list[dict[str, Any]] = []
+
+        # Access to Host list not available in bridge mode, API return error_code 'nodev'
+        if self.supports_hosts:
+            hosts_list_initialized = False
+            try:
+                fbx_devices.append(await self._api.lan.get_hosts_list())
+                hosts_list_initialized = True
+            except HttpRequestError as err:
+                m = re.search('Request failed \(APIResponse: (.+?)\)', str(err))
+                if m:
+                    json_str = m.group(1)
+                    try:                        
+                        json_resp = json.loads(json_str)
+                    except ValueError as ve:
+                        _LOGGER.error(
+                            "Failed to parse JSON '%s', error '%s'",
+                            json_str,
+                            ve,
+                        )
+
+                    if 'error_code' in json_resp and json_resp['error_code'] == 'nodev':
+                        # No need to retry, Host list not available
+                        self.supports_hosts = False
+                        hosts_list_initialized = True
+                        _LOGGER.info(
+                            "Host list is not available using bridge mode (%s)",
+                            json_resp['msg'],
+                        )
+
+                if not hosts_list_initialized:
```

Comment:
or simply "else" ?

And so we don't need `hosts_list_initialized`

Reply:
Not because in the case of `except ValueError as ve` I want to throw cause Exception

Reply:
Is it ok for you ?